docs/edgeql/expressions/parameters.rst

```diff
@@ -0,0 +1,173 @@
+.. _ref_eql_expr_params:
+
+Parameters
+==========
+
+:edb-alt-title: Query Parameters
+
+A parameter reference is used to indicate a value that is supplied externally
+to an EdgeQL expression.
+
+An example of query, that can be parametrized:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT Person { birth_date } FILTER .first_name = 'John';
+    {
+        Object {birth_date: <cal::local_date>'1971-01-01'},
+    }
+
+The equivalent parametrized query:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT Person { birth_date } FILTER .first_name = <str>$name;
+    Parameter <str>$name: John
+    {
+        Object {birth_date: <cal::local_date>'1971-01-01'},
+    }
+
+Note how the actual parameter value is asked by REPL in a separate prompt,
+after the whole query is accepted.
+
+While parameters are useful in REPL, their power is mostly when using language
+bindings. Example of using parameters in Python:
+
+.. code-block:: python
+
+    await connection.fetchone(
+        "SELECT Person { birth_date } FILTER .first_name = <str>$name",
+        name="John")
+
+Example of using parameters in JavaScript:
+
+.. code-block:: javascript
+
+    await connection.fetchOne(
+        "SELECT Person { birth_date } FILTER .first_name = <str>$name",
+        {name: "John"})
+
+
+Parameter Types
+---------------
+
+The following values can be turned into parameters:
+
+* Any :ref:`scalar type <ref_datamodel_scalar_types>`
+* :eql:type:`Array <array>` of any scalar type
+
+Parameter type is specified in the angular brackets in front of variable
+name, similarly to normal type cast:
+
+.. code-block:: edgeql
+
+    INSERT Person {
+        first_name := <str>$first_name,
+        age := <int32>$age,
+        interests := <array<str>>$interests,
+    }
+
+See :ref:`ref_eql_types` for more info on specifying types.
+
+
+Optional Parameters
+-------------------
+
+By default parameter types are ``REQUIRED`` which means that query will fail
+if parameter value is nil. You can use ``OPTIONAL`` keyword for optional
```

Comment:
```suggestion
if parameter value isn't provided. You can use the ``OPTIONAL`` keyword to denote optional
```

Reply:
"Isn't provided" is not true. We require `fetchone(..., param=None)` explicitly. And would fail before reaching postgres otherwise. So I couldn't come up with anything more clear than `nil`.